conn.go

```diff
@@ -989,6 +989,31 @@ func (c *Conn) requestHeader(apiKey apiKey, apiVersion apiVersion, correlationID
 	}
 }
 
+// ApiVersions return the list of API versions supported by the broker.
+func (c *Conn) ApiVersions() (versions []ApiVersionsV1, err error) {
```

Comment:
Is there a reason why we would need to call this from a program? If not we shouldn't export it.

Reply:
I would say that it depends if the strategy is to let the user defining which protocol version to use or if we do that automatically in the library.

Reply:
Let's do it automatically, kafka-go aims to be simple to use, this kind of implementation complexity is worth abstracting.